project_remaining_time_on_task/models/project.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+from openerp import api, fields, models
+
+
+class Task(models.Model):
+    _inherit = 'project.task'
+
+    has_contract_hours = fields.Boolean(string="Has contract hours",
+                                        compute="_compute_has_contract_hours")
+
+    remaining_contract_hours = fields.Float(
+        string="Remaining Contract Hours",
+        compute='_compute_remaining_hours_contract',
+        digits=(16, 2))
+
+    # @api.multi
+    # def _get_qty_invoiced(self):
+    #     self.ensure_one()
+    #     sale_order = self.env['sale.order'].search(
+    #         [('project_id', '=', self.analytic_account_id.id)])
+    #     return sum(line.qty_invoiced for line in sale_order.order_line)
+
+    @api.multi
+    def _get_qty_effective(self):
```

Comment:
I'd name this `_get_project_effective_qty`. 

Now that I think of it, you must make sure you count the hours on the same analytical account as the task you're calling from. Par exemple sur cette [tâche](https://gestion.coopiteasy.be/web#id=1318&view_type=form&model=project.task&action=479), j'ai pointé sur un compte "offert" qui ne doit pas être compté pour calculer les heures restantes sur le contrat.